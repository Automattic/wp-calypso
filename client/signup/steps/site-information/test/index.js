/**
 * @format
 * @jest-environment jsdom
 */

/**
 * External dependencies
 */
import React from 'react';
import { shallow } from 'enzyme';

/**
 * Internal dependencies
 */
import SignupActions from 'lib/signup/actions';
import { SiteInformation } from '../';

jest.mock( 'lib/signup/actions', () => ( {
	submitSignupStep: jest.fn(),
	saveSignupStep: jest.fn(),
} ) );

describe( '<SiteInformation />', () => {
	const defaultProps = {
		siteType: 'business',
		submitStep: jest.fn(),
		updateStep: jest.fn(),
		informationType: 'title',
		translate: x => x,
		siteInformation: { title: 'Ho ho ho!' },
		headerText: 'headerTextoidaliciously',
		formFields: [ 'title', 'address', 'phone' ],
		stepName: 'site-information',
	};

	afterEach( () => {
		SignupActions.submitSignupStep.mockReset();
		SignupActions.saveSignupStep.mockReset();
	} );

	test( 'should render', () => {
		const wrapper = shallow( <SiteInformation { ...defaultProps } /> );
		expect( wrapper ).toMatchSnapshot();
	} );

	test( 'should update field attributes based on the informationFields array values', () => {
		const wrapper = shallow(
			<SiteInformation { ...defaultProps } formFields={ [ 'phone', 'address' ] } />
		);
		const siteInformationContent = shallow( wrapper.instance().renderContent() );
		const inputFields = siteInformationContent.find( 'FormTextInput' );
		expect( inputFields ).toHaveLength( 2 );
		expect( inputFields.get( 0 ).props.name ).toEqual( 'phone' );
		expect( inputFields.get( 1 ).props.name ).toEqual( 'address' );
	} );

	test( 'should call `submitStep()` from `handleSubmit()`', () => {
		const wrapper = shallow( <SiteInformation { ...defaultProps } /> );
		wrapper.instance().handleSubmit( {
			preventDefault: () => {},
		} );
		expect( defaultProps.submitStep ).toHaveBeenCalledWith( defaultProps.siteInformation );
	} );

	test( 'should call `updateStep()` from `handleInputChange()`', () => {
		const wrapper = shallow( <SiteInformation { ...defaultProps } /> );
		wrapper.instance().handleInputChange( {
			currentTarget: { name: 'title', value: 'Tada!!!' },
		} );
		expect( defaultProps.updateStep ).toHaveBeenCalledWith( 'title', 'Tada!!!' );
	} );
} );
