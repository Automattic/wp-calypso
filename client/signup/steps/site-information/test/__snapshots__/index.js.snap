// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SiteInformation /> should render 1`] = `
<Localized(StepWrapper)
  fallbackHeaderText="headerTextoidaliciously"
  fallbackSubHeaderText="subHeaderTextoidaliciously"
  headerText="headerTextoidaliciously"
  stepContent={
    <div
      className="site-information__wrapper is-single-fieldset"
    >
      <Card
        highlight={false}
        tagName="div"
      >
        <form>
          <div
            className="site-information__field-control site-information__title"
          >
            <FormFieldset>
              <Localized(FormLabel)
                htmlFor="title"
              >
                What is the name of your business?
                <InfoPopover
                  autoRtl={true}
                  className="site-information__info-popover"
                  icon="info-outline"
                  iconSize={18}
                  position="top"
                >
                  We'll use this as your site title. Don't worry, you can change this later.
                </InfoPopover>
              </Localized(FormLabel)>
              <FormTextInput
                autoFocus={true}
                id="title"
                maxLength={100}
                name="title"
                onChange={[Function]}
                placeholder="E.g., Vail Renovations"
                value="Ho ho ho!"
              />
              <Button
                onClick={[Function]}
                primary={true}
                type="submit"
              >
                <t
                  icon="arrow-right"
                  size={24}
                />
              </Button>
            </FormFieldset>
          </div>
          <div
            className="site-information__field-control site-information__address"
          >
            <FormFieldset>
              <Localized(FormLabel)
                htmlFor="address"
              >
                Address
                <InfoPopover
                  autoRtl={true}
                  className="site-information__info-popover"
                  icon="info-outline"
                  iconSize={18}
                  position="top"
                >
                  Where can people find your business?
                </InfoPopover>
              </Localized(FormLabel)>
              <FormTextInput
                autoFocus={false}
                id="address"
                maxLength={75}
                name="address"
                onChange={[Function]}
                placeholder="E.g., 60 29th St, San Francisco, CA 94110"
                value=""
              />
              <Button
                onClick={[Function]}
                primary={true}
                type="submit"
              >
                <t
                  icon="arrow-right"
                  size={24}
                />
              </Button>
            </FormFieldset>
          </div>
          <div
            className="site-information__field-control site-information__phone"
          >
            <FormFieldset>
              <Localized(FormLabel)
                htmlFor="phone"
              >
                Phone number
                <InfoPopover
                  autoRtl={true}
                  className="site-information__info-popover"
                  icon="info-outline"
                  iconSize={18}
                  position="top"
                >
                  How can people contact you?
                </InfoPopover>
              </Localized(FormLabel)>
              <FormTextInput
                autoFocus={false}
                id="phone"
                maxLength={50}
                name="phone"
                onChange={[Function]}
                placeholder="E.g., (555) 555-5555"
                value=""
              />
              <Button
                onClick={[Function]}
                primary={true}
                type="submit"
              >
                <t
                  icon="arrow-right"
                  size={24}
                />
              </Button>
            </FormFieldset>
          </div>
        </form>
      </Card>
    </div>
  }
  stepName="site-information"
  subHeaderText="subHeaderTextoidaliciously"
/>
`;
